cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramRunStatusMonitorService.java

```diff
@@ -169,4 +181,32 @@ protected long runTask() {
       return terminateTime != null && terminateTime <= currentTimeInSecs;
     };
   }
+
+  private void emitForceTerminatedRunsMetrics(ProgramRunId programRunId, RunRecordDetail recordedRunRecord,
+                                              String metricName) {
+    Optional<ProfileId> profile = SystemArguments.getProfileIdFromArgs(programRunId.getNamespaceId(),
+                                                                       recordedRunRecord.getSystemArgs());
+    Map<String, String> additionalTags = getAdditionalTagsForProgramMetrics(recordedRunRecord);
+    profile.ifPresent(profileId -> emitProfileMetrics(programRunId, profileId, metricName, additionalTags));
+  }
+
+  private Map<String, String> getAdditionalTagsForProgramMetrics(RunRecordDetail runRecord) {
+    Map<String, String> additionalTags = new HashMap<>();
+    // don't want to add the tag if it is not present otherwise it will result in NPE
+    additionalTags.computeIfAbsent(Constants.Metrics.Tag.PROVISIONER,
+                                   provisioner -> SystemArguments.getProfileProvisioner(runRecord.getSystemArgs()));
+    return additionalTags;
+  }
+
+  private void emitProfileMetrics(ProgramRunId programRunId, ProfileId profileId, String metricName,
+                                  Map<String, String> additionalTags) {
+    Map<String, String> tags = ImmutableMap.<String, String>builder()
+      .put(Constants.Metrics.Tag.PROFILE_SCOPE, profileId.getScope().name())
+      .put(Constants.Metrics.Tag.PROFILE, profileId.getProfile())
+      .put(Constants.Metrics.Tag.NAMESPACE, programRunId.getNamespace())
```

Comment:
these tags are different than the ones used for the stopping metric. Is that intentional? 

For example, program is not a tag here, but it is in the other place. This means you can get stopping time for a particular program, but not the number of force kills.

Reply:
Good catch. Added the program tag. The runId tag is the only difference since it won't apply to total number of force terminated runs as it is an aggregate.